Tools/ros2/ardupilot_dds_tests/test/ardupilot_dds_tests/test_rally_point.py

```diff
@@ -0,0 +1,204 @@
+        try:
+            self.send_thread.stop()
+        except:
+            self.send_thread = threading.Thread(target=self.process_send, args=[rally])
+            self.send_thread.start()
+
+    def process_compare(self, rally):
+        while True:
+            response = self._get_rally()
+            if response is not None:
+                if response.success:
+                    print (f"{response.rally.point.latitude} - {rally.point.latitude}")
+                    if (abs(response.rally.point.latitude - rally.point.latitude) < 1e-7
+                        and abs(response.rally.point.longitude - rally.point.longitude) < 1e-7
+                        and response.rally.altitude_frame == rally.altitude_frame):
+                        self.get_rally_event.set()
+                    return
+            time.sleep(1)
+        
+    def get_and_compare_rally(self, rally):
+        try:
+            self.get_thread.stop()
+        except:
+            self.get_thread = threading.Thread(target=self.process_compare, args=[rally])
+            self.get_thread.start()    
+
+@launch_pytest.fixture
+def launch_sitl_copter_dds_serial(sitl_copter_dds_serial):
+    """Fixture to create the launch description."""
+    sitl_ld, sitl_actions = sitl_copter_dds_serial
+
+    ld = LaunchDescription(
+        [
+            sitl_ld,
+            launch_pytest.actions.ReadyToTest(),
+        ]
+    )
+    actions = sitl_actions
+    yield ld, actions
+
+@launch_pytest.fixture
+def launch_sitl_copter_dds_udp(sitl_copter_dds_udp):
+    """Fixture to create the launch description."""
+    sitl_ld, sitl_actions = sitl_copter_dds_udp
+
+    ld = LaunchDescription(
+        [
+            sitl_ld,
+            launch_pytest.actions.ReadyToTest(),
+        ]
+    )
+    actions = sitl_actions
+    yield ld, actions
+    
+@pytest.mark.launch(fixture=launch_sitl_copter_dds_udp)
+def test_dds_udp_rally(launch_context, launch_sitl_copter_dds_udp):
+    """Test Rally Points with AP_DDS."""
+    _, actions = launch_sitl_copter_dds_udp
+    micro_ros_agent = actions["micro_ros_agent"].action
+    mavproxy = actions["mavproxy"].action
+    sitl = actions["sitl"].action
+
+    # Wait for process to start.
+    process_tools.wait_for_start_sync(launch_context, micro_ros_agent, timeout=2)
+    process_tools.wait_for_start_sync(launch_context, mavproxy, timeout=2)
+    process_tools.wait_for_start_sync(launch_context, sitl, timeout=2)
+
+    rclpy.init()
+    time.sleep(5)
+    try:
+        node = RallyControl()
+        node.clear_list()
+        clear_flag = node.clear_rally_event.wait(20)
+        assert clear_flag, "Could not clear"
+        node.send_rally(RALLY_0)
+
+        send_flag = node.add_rally_event.wait(10)
+        assert send_flag, "Could not send Rally"
+        
+        node.get_and_compare_rally(RALLY_0)
+        send_flag = node.get_rally_event.wait(10)
+        assert send_flag, "Wrong Rally point back"        
+    except Exception as e:
```

Comment:
Remove exception handling. Don't catch them, especially like this. It's going to swallow all issues in the code, including stuff that should fail the test.

Reply:
that was a leftover of my debugging, thanks for pointing it out.